src/key/local.rs

```diff
@@ -87,25 +73,28 @@ impl LocalCreator {
         );
         Self::validate_path(self.path.as_path())?;
 
-        let mut file = File::create(self.path.join(&self.name).as_path())
+        let mut file = File::create(self.path.as_path())
             .map_err(|e| LocalKeyCreationError::UnableToCreatePrivateKeyFile(e.to_string()))?;
         file.write_all(key)
             .map_err(|e| LocalKeyCreationError::UnableToWritePrivateKey(e.to_string()))?;
         Ok(())
     }
 
     fn validate_path(path: &Path) -> Result<(), LocalKeyCreationError> {
-        if !path.exists() {
-            return Err(LocalKeyCreationError::InvalidPath(String::from(
-                "local key path does not exist",
-            )));
+        // Filename should not exist already
+        if path.exists() {
+            Err(LocalKeyCreationError::InvalidPath(String::from(
+                "local key path already exists",
+            )))
         }
-        if path.is_file() {
-            return Err(LocalKeyCreationError::InvalidPath(String::from(
-                "local key path needs to be a directory",
-            )));
+        // The parent directory must exist and be a directory
+        else if !path.parent().is_some_and(|p| p.exists() && p.is_dir()) {
+            Err(LocalKeyCreationError::InvalidPath(String::from(
+                "local key path parent directory does not exist or is not a directory",
+            )))
```

Comment:
Have you thought about creating the parent directories if they don't exist?
Do you think it would be a sane default to do it or a wrong behaviour?
I think it could be great to remove the `mkdir` from the bash script. I don't see any other advantage. The disadvantage is that we might create folders the user didn't want to create.

https://doc.rust-lang.org/beta/std/fs/fn.create_dir_all.html

Reply:
Hmm yeah I think it could be a good default. I'll add it.

Reply:
Done in https://github.com/newrelic/newrelic-auth-rs/pull/52/commits/8c880aa30a493e5a2ddae13fa1384b359839e71a!